feat: cancel last action functionnality

haangeline opened the pull request:
enables to cancel the last action when drawing

haangeline committed:
feat: cancel last action functionnality

haangeline committed:
fix: removed duplicated code ; removed useless config modifications ; updated lock files

haangeline committed:
Merge branch 'ArchireportTeam:main' into feat/cancelLastAction

mlecoq commented on src/components/DrawCore/index.tsx:
```diff
@@ -254,1044 +255,1098 @@ const DrawCore = React.forwardRef<
+    const updateScreenStates = useCallback((done: DrawItem[]) => {
+      const newScreenStates = screenStates;
+      newScreenStates.push(done);
+      setScreenStates(newScreenStates);
```
instead of using screenStates (and having to add it in hook dependencies), please use a method in setter

```
setScreenStates(previous => {
   previous.push(done);
   return previous;
});
```

mlecoq commented on src/components/DrawCore/index.tsx:
```diff
@@ -254,1044 +255,1098 @@ const DrawCore = React.forwardRef<
+    const deleteScreenStates = useCallback((indice: number) => {
+      const newScreenStates = screenStates;
+      newScreenStates.splice(indice, 1);
+      setScreenStates(newScreenStates);
```
instead of using screenStates (and having to add it in hook dependencies), please use a method in setter

```
setScreenStates(previous => {
   previous.splice(indice, 1);
   return previous;
});
```

mlecoq commented on src/components/DrawCore/index.tsx:
```diff
@@ -254,1044 +255,1098 @@ const DrawCore = React.forwardRef<
+    const initialItem = useSharedValue<DrawItem | null>(null);
 
-  const [doneItems, setDoneItems] = useState<DrawItem[]>([]);
+    const [doneItems, setDoneItems] = useState<DrawItem[]>([]);
```
You should remove `doneItems` : `doneItems` should be the same as the last element of `screenStates` so you could use `screenStates.at(-1)` as `doneItems`

haangeline committed:
feat: cancel color and strokeWidth changes

haangeline committed:
refactor: doneItems and screenStates states are now in useReducer drawStates

added onColorStrokeChange callback ; deleted strokeChange and colorChange states
